modules/ctl/src/main/scala/low/git.scala

```diff
@@ -0,0 +1,58 @@
+package gem.ctl
+package low
+
+import scalaz._, Scalaz._
+import scala.util.matching.Regex
+
+import gem.ctl.free.ctl._
+import io._
+
+/** Low-level constructors for `CtlIO` operations related to git. */
+object git {
+
+  case class Commit(hash: String)
+  object Commit {
+    implicit val OrderCommit: Order[Commit] = Order.orderBy(_.hash)
+  }
+
+  case class Tag(tag: String)
+
+  // Yields most recent tag matching the given pattern, if any
+  def mostRecentTag(pattern: Regex): CtlIO[Option[Tag]] =
+    shell("git", "tag").require {
```

Comment:
This is a tough call. I wanted to keep it as simple as possible (like a shell script) so you can look at the code and the `-v` log output and know what's going on. We can switch to a library if there end up being problems, but there's an awful lot of automation out there that depends on git output so I'm not *that* worried about it. But yeah it's a potential concern. Docker as well.